### PR TITLE
Simplify the code to generate font substitution information

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

**Comment by calixteman** [comment]
Unit tests are ok locally and I think it really simplifies the code.
If you see something I can improve please tell me.

File: src/core/font_substitutions.js [hunk 1]
```diff
@@ -238,13 +208,15 @@ const substitutionMap = new Map([
   [
     "ArialBlack-BoldItalic",
     {
-      alias: "ArialBlack-Italic",
+      alias: {
+        name: "ArialBlack-Italic",
+      },
```
Comment (Snuffleupagus) [inline src/core/font_substitutions.js:213]: Shouldn't this be formatted as follows now?
```suggestion
      alias: "ArialBlack-Italic",
```

File: src/core/font_substitutions.js [hunk 2]
```diff
@@ -333,38 +287,96 @@ const substitutionMap = new Map([
...
+    default:
+      if (!style) {
+        return "";
+      }
+      if (style.weight === "bold") {
+        return "Bold";
+      }
+      if (style.style === "italic") {
+        return "Italic";
+      }
```
Comment (Snuffleupagus) [inline src/core/font_substitutions.js:307]: Given how this is a fallback code-path, we could perhaps shorten the code a little bit?
```suggestion
      if (style?.weight === "bold") {
        return "Bold";
      }
      if (style?.style === "italic") {
        return "Italic";
      }
```

File: src/core/font_substitutions.js [hunk 3]
```diff
@@ -333,38 +287,96 @@ const substitutionMap = new Map([
...
+      fallbackInfo,
+      src,
+      localFontPath,
+      useFallback,
+      /* usePath = */ usePath && !path,
+      append
+    );
+    if (!result.ultimate) {
+      result.ultimate = fallbackUltimate;
+    }
```
Comment (Snuffleupagus) [inline src/core/font_substitutions.js:373]: Now that we're able to use modern JS also in the worker-thread, how about the following?
```suggestion
    result.ultimate ||= fallbackUltimate;
```

**Review by Snuffleupagus — APPROVED** [review]
r=me, with a few more comments and passing tests; thank you!

**Pull request merged**